rust/src/main.rs

```diff
@@ -29,13 +29,27 @@ use std::path::{Path, PathBuf};
 use xi_core_lib::XiCore;
 use xi_rpc::RpcLoop;
 
-const XI_DIRECTORY: &str = "xi-core";
+const XI_LOG_DIR: &str = "xi-core";
 const XI_LOG_FILE: &str = "xi-core.log";
 
-fn get_logging_directory() -> Result<PathBuf, io::Error> {
+fn get_logfile_directory_name() -> String {
+    match std::env::var("XI_LOG_DIR") {
```

Comment:
so we've moved away from using env vars for stuff like this; for instance we used to do this for config files (and might still support it) but now we set those directories through the RPC system.

We can't do that here, but I think it makes more sense to just pass args to the executable directly, unless you have a particular reason for preferring env vars?